src/scheduler/workers/main.ts

```diff
@@ -0,0 +1,264 @@
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "POST_ONE_DAY", "sendOneDayPostEmail");
+        }
+        break;
+      case annualReviewKeyDates?.START:
+        // email posts to notify of annual review start
+        isEmailSent = isEmailSentBefore(audit as Audit, "sendStartedPostEmail");
+        if (!isEmailSent) {
+          await processPostEmailsForList(list, "START", "sendStartedPostEmail");
+        }
+
+        // update ListItem.isAnnualReview if today = the START milestone date
+        // email providers to notify of annual review start
+        isEmailSent = isEmailSentBefore(listItemAudit as Audit, "sendStartedProviderEmail");
+        if (!isEmailSent) {
+          const updatedListItems = await updateIsAnnualReviewForListItems(listItemsForList, list);
+          const currentDateString = formatDate(addDays(parseDate(todayDateString), 42));
+          await processProviderEmailsForListItems(list, updatedListItems, "START", currentDateString);
+        }
+        break;
+    }
+  }
+}
+
+export async function updateIsAnnualReviewForListItems(listItems: ListItemWithHistory[], list: List): Promise<ListItemWithHistory[]> {
+  if (listItems.length === 0) {
+    logger.info(`No List items found for list ${list.id}`);
+    return [];
+  }
+  const updatedListItems: Result<ListItemWithHistory[]> = await updateIsAnnualReview(
+    list,
+    listItems,
+    ListItemEvent.ANNUAL_REVIEW_STARTED,
+    "startAnnualReview",
+    AuditEvent.REMINDER
+  );
+
+  if (updatedListItems.error) {
+    logger.info(`Unable to update list items and email providers for annual review start`);
+    return [];
+  }
+  if (!updatedListItems.result?.length) {
+    logger.info(`No list items were updated for annual review start`);
+    return [];
+  }
+  if (updatedListItems.result.length < listItems.length) {
+    const listItemsNotUpdated = listItems.filter((listItem) => {
+      // @ts-ignore
+      return !updatedListItems.result
+        .map((updatedListItem) => updatedListItem.id)
+        .includes(listItem.id);
+    });
+    logger.error(
+      `List items ${listItemsNotUpdated.map((listItem) => listItem.id)} were not updated for annual review start`
+    );
+  }
+  return updatedListItems.result;
+}
+
+async function processAnnualReview(): Promise<void> {
+  const listResult = await findListsWithCurrentAnnualReview();
+  logger.debug(`Found ${listResult.result?.length} Lists with current annual review populated`);
+
+  // validate list results
+  if (!listResult.result?.length) {
+    logger.info(`No lists found with current annual review populated`);
+    return;
+  }
+  const listsWithCurrentAnnualReview = listResult.result;
+  const listItemIds: number[] = [];
+  listsWithCurrentAnnualReview.forEach((list: List) => {
+    if (list.jsonData?.currentAnnualReview?.eligibleListItems.length) {
+      list.jsonData?.currentAnnualReview?.eligibleListItems.map((listItemId) => listItemIds.push(listItemId));
```

Comment:
you can remove the if statement. you can call methods conditionally also.

list.jsonData?.currentAnnualReview?.eligibleListItems.map?.((listItemId) => listItemIds.push(listItemId));

I think you are trying to do a flatmap?

```
lists.flatMap(list => {
  return list.items.map(item => item.id)
})
```
regular map will return you something like `[[1, 2, 3], [4, 5, 6]]`, flapMap will give you `[1, 2, 3, 4, 5, 6]`.

although I think this can be done by prisma query?

Reply:
yep, I was looking for something like flatMap, thx.  After a bit of googling, I don't think prisma queries support flattening of json array result data so it needs to be flattened post querying.

Reply:
Sorry, I didn't mean flat mapping can be done by prisma. but perhaps a single prisma query can do what the filtering/flatmapping is trying to achieve to begin with. 

So it looks like you're 

1. looking for lists that have an annualReview object
2. iterating over the lists to find the eligible list items
3. reconciling list and list items via maps and filter
4. processing list and the eligible items 


Could you do something like 

```.ts
await prisma.listItem.findMany({
    where: {
      list: {
        jsonData: {...},
      },
    },
  });
```

or 

```.ts
 await prisma.list.findMany({
    where: {
      jsonData: {...},
    },
    include: {
      items: {
        where: {
          list: {
            jsonData: {...},
          },
        },
      },
    },
  });
```

Reply:
I reckon hoisting into the list query is doable - will check it out closer....